### PR TITLE
make supercell test case threadsafe

### PR DESCRIPTION
**Pull request by mt5555**
minor bugfixes for the standlone HOMME supercell test case

thread safe initialization
bug fix for restart
turn on lustre hints on Cori-KNL
[BFB]

### CODE REVIEW

File: components/homme/src/test_mod.F90 [hunk 1]
```diff
@@ -77,7 +77,8 @@ subroutine set_test_initial_conditions(elem, deriv, hybrid, hvcoord, tl, nets, n
   endselect
 
   !initial conditions for initial run, runtype=0
-  if (runtype == 0) then
+  ! also does other test case setup.
```
Comment (oksanaguba) [inline components/homme/src/test_mod.F90]: Does this init code run for restart runs? What does the comment above, 'other test case', mean?

File: components/homme/src/test_src/dcmip12_wrapper.F90 [hunk 2]
```diff
@@ -634,7 +634,7 @@ subroutine set_hybrid_coefficients(hv, hybrid, eta_t, c)
     tmp        = max( (hv%etai(k)-eta_c)/(1.0-eta_c), 0.0_rl)
     hv%hybi(k) = tmp**c
     hv%hyai(k) = hv%etai(k) - hv%hybi(k)
-    if(hybrid%par%masterproc) write(*,'(i4,a,f18.15,a,f18.15,a,f18.15)') &
+    if(hybrid%masterthread) write(*,'(i4,a,f18.15,a,f18.15,a,f18.15)') &
```
Comment (oksanaguba) [inline components/homme/src/test_src/dcmip12_wrapper.F90]: Why change masterproc to masterthread here? Without looking, i'd think this is not intended?

File: components/homme/src/test_mod.F90 (hunk 1 above)
Comment (mt5555) [reply components/homme/src/test_mod.F90]: i meant the other setup in addition to the initial condition.  some test cases (like supercell) initialize arrays that need to be done both on new runs and restart runs.

File: components/homme/src/test_src/dcmip12_wrapper.F90 (hunk 2 above)
Comment (mt5555) [reply components/homme/src/test_src/dcmip12_wrapper.F90]: it's in a threaded region, but we only want one thread to print this message.

**Pull request merged**